### PR TITLE
✨ Add event filter for HelmReleaseProxy when it is paused

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**: Adds an event filter for HelmReleaseProxy to make it consistent with HelmChartProxy. This will allow reconciliation to be paused when the pause annotation is applied.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/helmreleaseproxy/helmreleaseproxy_controller.go [hunk 1]
```diff
@@ -37,22 +37,27 @@ import (
...
 func (r *HelmReleaseProxyReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	_ = ctrl.LoggerFrom(ctx)
+	log := ctrl.LoggerFrom(ctx)
 
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
 		For(&addonsv1alpha1.HelmReleaseProxy{}).
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
+		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(log, r.WatchFilterValue)).
 		// Watches(
```
Comment (CecileRobertMichon) [inline controllers/helmreleaseproxy/helmreleaseproxy_controller.go:61]: not part of this PR but why is this block commented out? if it's not used consider removing it

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm

@Jont828 let's add a quick PR description?

**Comment by Jont828** [comment]
@CecileRobertMichon I think we need the /lgtm here as well.
